### PR TITLE
Updated Featured Cars section along with Unit test And Navigation bar updated

### PR DESCRIPTION
**Pull request by kboodi**
Added Unit testing to both featured cars and car card components.
Also changed background color of arrow in pagination.
Updated the timer to 5 second in auto rotating feature.

image

image

Navigation bar changes:
Logo changed and made it clickable
added All cars Link to nav bar which routes to car listing page.

View All link changed
image

Jira Ticket links:
https://abhirams.atlassian.net/browse/P3ECWFC-34
https://abhirams.atlassian.net/browse/P3ECWFC-72
https://abhirams.atlassian.net/browse/P3ECWFC-73

### CODE REVIEW

**Comment by pashedin** [comment]
wasn't it our intention to have the nigation buttons look like this?
Screenshot 2022-04-27 at 5 20 29 PM

**Comment by pashedin** [comment]
I also notice the sub-text has a different font color in designs

**Review by pashedin — CHANGES_REQUESTED** [review]
design changes first and foremost

File: src/assets/Arrow.svg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<svg width="32" height="20" viewBox="0 0 32 20" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path d="M32 9.84785L22.6975 0L22.2544 0.412282L30.8793 9.54283L22.591 9.46374L18.4447 5.12262L18.0039 5.53732L21.7464 9.45568L0.00583992 9.24821L0 9.85098L21.8064 10.0591L18.0008 14.1681L18.4479 14.576L22.6241 10.0669L30.9061 10.1459L22.2513 19.5945L22.7007 20L32 9.84785Z" fill="black"/>
```
Comment (pashedin) [inline src/assets/Arrow.svg:2]: Please reuse the Right Cross Icon, that's already being used in the carousel

**Comment by pashedin** [comment]
Also I see there are Css changes for Search bar, would love to see a screenshot of that as well. Please update the PR description with the same

File: src/components/FeatureCars/__test__/FeaturedCars.test.tsx [hunk 2]
```diff
@@ -1,9 +1,45 @@
...
+    <BrowserRouter>
+      <Provider store={store}>
+        <FeaturedCars carTypeIndex={0} />
+      </Provider>
+    </BrowserRouter>
+  );
+};
+
+describe("featuredcars component testing", () => {
+  it("should renders features viewall text", () => {
```
Comment (pashedin) [inline src/components/FeatureCars/__test__/FeaturedCars.test.tsx:19]: nice

File: src/components/Navbar/TopNavbar.tsx [hunk 3]
```diff
@@ -2,24 +2,36 @@ import React from "react";
...
-        <h5>TR</h5>
-        <p>adeurcar</p>
-      </Navbar.Brand>
-      <Nav className="container2">
-        <Link to="/cars" className="nav-link">
-          New Cars
+    <div className="layout--header--main--container">
+      <nav className="navbar navbar-expand-lg navbar-dark bg-dark layout--header">
+        <Link to={"/"} className="navbar-brand layout--header--logo">
+          <img src={Logo} alt="Logo" />
```
Comment (pashedin) [inline src/components/Navbar/TopNavbar.tsx]: alt should not be vague, it should describe what the image is representing, and what it means. 
So a more relevant alt text would be `xtreme car logo`

File: src/components/Navbar/__test__/TopNavbar.test.tsx [hunk 4]
```diff
@@ -1,9 +1,38 @@
...
+    const ProfileElement = screen.getByText(/Profile/i);
+    expect(ProfileElement).toBeInTheDocument();
+  });
+
+  it("should render logotext in the Navbar", () => {
+    render(<MockNavbar />);
+    const LogoElement = screen.getByText(/XTREME CARS/i);
+    expect(LogoElement).toBeInTheDocument();
+  });
+
```
Comment (pashedin) [inline src/components/Navbar/__test__/TopNavbar.test.tsx:63]: we also need to include tests that, check that the links are taking the users to the right page

File: src/components/Searchbar/Searchbar.css [hunk 5]
```diff
@@ -0,0 +1,32 @@
+.searchbar {
```
Comment (pashedin) [inline src/components/Searchbar/Searchbar.css]: css files?
@Pratishah 
remove this, as well as the corresponding css.map

**Review by pashedin — CHANGES_REQUESTED** [review]

File: src/components/FeatureCars/FeaturedCars.tsx [hunk 6]
```diff
@@ -1,12 +1,14 @@
 import React, { useCallback, useEffect, useState } from "react";
-import { Pagination } from "react-bootstrap";
+import "./FeaturedCars.scss";
+import CarCard from "../CarCard/CarCard";
+import { Button, Pagination } from "react-bootstrap";
 import { ArrowRight } from "react-bootstrap-icons";
```
Comment (pashedin) [inline src/components/FeatureCars/FeaturedCars.tsx]: no need for this now right?

Comment (pashedin) [reply src/components/FeatureCars/FeaturedCars.tsx]: View All should also have the right designer icon from figma

Comment (kboodi) [reply src/components/FeatureCars/FeaturedCars.tsx]: Made this change

File: src/components/CarCard/CarCard.tsx [hunk 7]
```diff
@@ -1,26 +1,40 @@
...
+      <Card className="cardContainer ">
+        <Card.Img
+          variant="top"
+          data-testid="carimg"
+          src={carDetails.image}
+          height={200}
+        />
+        <Card.Body>
+          <Card.Title data-testid="cardetail">{carDetails.name}</Card.Title>
+          <Card.Text className="font-styles " data-testid="cardprice">
```
Comment (pashedin) [inline src/components/CarCard/CarCard.tsx]: what kind of class name is font-styles? 😆 
Please name it appropriately from one of these if possible
This is available in figma file, you can go there by choosing `Components` in the nav bar menu on the left
Screenshot 2022-04-29 at 11 51 35 AM

File: src/App.test.tsx [hunk 8]
```diff
@@ -1,9 +1,18 @@
...
+      <App />
+    </BrowserRouter>
+  );
+};
+
+xtest("renders learn react link", () => {
+  render(<Mockapp />);
   const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+  expect(linkElement).not.toBeInTheDocument();
```
Comment (pashedin) [inline src/App.test.tsx]: no need to maintian stupid test cases jus to pass the tests, remove them if the test is unnecessary

File: src/assets/Arrow.svg (hunk 1 above)
Comment (pashedin) [reply src/assets/Arrow.svg:2]: ☝️ remove/delete this icon and reuse the icon that's being used in the carousel

Comment (pashedin) [reply src/assets/Arrow.svg:2]: Also remove the Union.jpg file, use only the svg files

File: src/components/FeatureCars/FeaturedCars.tsx (hunk 6 above)
Comment (pashedin) [reply src/components/FeatureCars/FeaturedCars.tsx]: you would also need to remove the import from `react-bootstrap-icon`

**Review by pashedin — CHANGES_REQUESTED** [review]

File: public/index.html [hunk 9]
```diff
@@ -2,7 +2,7 @@
 <html lang="en">
   <head>
     <meta charset="utf-8" />
-    <link rel="icon" href="%PUBLIC_URL%/car-logo.jpeg" />
+    <link rel="icon" href="%PUBLIC_URL%/logo.svg" />
     <!-- <link rel="apple-touch-icon" href="../src/Assets/car-logo.jpeg" /> -->
```
Comment (pashedin) [inline public/index.html]: remove the comment if not being used

File: src/pages/CarsList/CarsList.tsx [hunk 10]
```diff
@@ -18,7 +18,7 @@ function CarsList() {
   useEffect(() => {
     window.scrollTo(0, 0);
   }, [location]);
-  
+
   const [carType, SetCarType] = useState<string>(
     ["sedan", "SUV", "hatchback", "coupe"].includes(
       JSON.parse(JSON.stringify(searchtext)).searchtext as string
```
Comment (pashedin) [inline src/pages/CarsList/CarsList.tsx]: Why do we need this?
`JSON.parse(JSON.stringify(searchtext))`

Please remove such usages from everywhere. stringify and parse are expensive operations, and there is little sense in doing both in same line.

**Review by pashedin — CHANGES_REQUESTED** [review]
some more please

File: src/components/FeatureCars/FeaturedCars.tsx [hunk 11]
```diff
@@ -73,10 +75,10 @@ function FeatureCars(props: any) {
       <div>
         <Link to="/cars" className="viewLink">
           View All{"  "}
-          <ArrowRight />
+          <img src={CrossArrow} alt="cross arrow" />
         </Link>
       </div>
-      <div className="cardslist">
+      <div className="cardslist" data-testid="carlist">
```
Comment (pashedin) [inline src/components/FeatureCars/FeaturedCars.tsx]: `cards-list`

File: src/components/CarCard/CarCard.tsx (hunk 7 above)
Comment (kboodi) [reply src/components/CarCard/CarCard.tsx]: card design has been modified including font size and color along with shadow.
Added box-shadow on hover only.
image

on hover:
image

File: public/index.html [hunk 12]
```diff
@@ -2,27 +2,28 @@
...
     <meta
       name="description"
       content="Web site created using create-react-app"
     />
-    <!-- <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" /> -->
-    <!--
-      manifest.json provides metadata used when your web app is installed on a
-      user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
-    -->
+    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
```
Comment (pashedin) [inline public/index.html]: This is logo192 so essentially with this line of code just for apple users, in some case they would see the react laog, while other people would continue to see the regular xtreme cars logo
logo192

File: src/components/CarCard/CarCard.tsx (hunk 7 above)
Comment (pashedin) [reply src/components/CarCard/CarCard.tsx]: why does the font weight of sub text change on hover?

**Review by pashedin — CHANGES_REQUESTED** [review]

**Review by pashedin — DISMISSED** [review]
Finally the hard work paid off 😄

**Comment by pashedin** [comment]
whoopsie resolve the conflicts please

**Review by pashedin — APPROVED** [review]

**Pull request merged**